dev: address a few paper cuts

irfansharif opened the pull request:
See individual commits, works through a few of the known paper cuts with dev/bazel (#75453).

irfansharif committed:
dev: get rid of --remote-cache

We set up a persistent bazel-remote cache process now through `dev
cache`, we no longer need this flag. We're also suggesting placing
these flags in appropriate bazelrcs instead.

Release note: None

irfansharif committed:
dev: symlink build --short result to `cockroach`

This is what the Makefile does, and what folks expect.

Release note: None

irfansharif committed:
dev: skip running tests as part of `dev bench`

Slipped in as part of #76189. Before this patch we were unintentionally
running all unit tests in whatever package was targeted. To repro, try:

    dev bench pkg/server -f='BenchmarkSetupSpanForIncomingRPC'

Release note: None

irfansharif committed:
dev: disable sharding when a filter is specified

For sharded test packages, it doesn't make much sense to spawn multiple
test processes that don't end up running anything. Default to running
things in a single process if a filter is specified. To compare, try the
following before/after this commit:

  dev test pkg/server -f=TestClusterVersionUpgrade -v

Release note: None

irfansharif committed:
dev: support --show-sql for logic tests

It's a fairly common flag to use.

Release note: None

irfansharif committed:
dev: support --count, --stress{,-args} for testlogic

Unify a few codepaths while here.

Release note: None

irfansharif committed:
dev: bump release version

Pick up changes in preceding commits.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/76415)
<!-- Reviewable:end -->

irfansharif committed:
bazel: import .bazelrc.user only after the defaults

It's closer to what folks would expect with per-user overrides.

Release note: None

irfansharif committed:
dev: introduce --test-args for go test binaries

It's useful to be able to pass arguments down to the go test binary
directly. This commit introduces a top-level flag to do just that. Now
we can do the following sort of thing:

    dev bench pkg/bench -f='BenchmarkTracing/1node/scan/trace=off' \
      --test-args '-test.memprofile=mem.out -test.cpuprofile=cpu.out'

Release note: None

RaduBerinde reviewed (commented):
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> Thanks for addressing all these, very cool!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/76415#-:-MveVhv97uZSDdCUNGVk:b-g96b1t)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @irfansharif and @rickystewart)

___
*[-- commits, line 20 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/76415#-MveUl3f1yoFHo2RAbT9):*
[supernit] we were
___
*[go.mod, line 166 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/76415#-MveVW0q0GBocCfBygRT:-MveVW0q0GBocCfBygRU:b7ga9yw) ([raw file](https://github.com/cockroachdb/cockroach/blob/12b5901fd5529c7358908c0b15d691f63dca7061/go.mod#L166)):*
> ```XML
> )
> 
> require github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510
> ```

FWIW, we use `github.com/alessio/shellescape` elsewhere


<!-- Sent from Reviewable.io -->

irfansharif reviewed (commented):
TFTR!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/76415#-:-MvejPkQ-ZTj0YtQ53Eg:b-oduykh)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @RaduBerinde and @rickystewart)

___
*[go.mod, line 166 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/76415#-MveVW0q0GBocCfBygRT:-MvehsJu0snYQiBk9yx3:b-gkod4z) ([raw file](https://github.com/cockroachdb/cockroach/blob/12b5901fd5529c7358908c0b15d691f63dca7061/go.mod#L166)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

FWIW, we use `github.com/alessio/shellescape` elsewhere
</blockquote></details>

I was using it for the underlying shell lexer to step through tokens (there are two uses, both for datadriven and for --test-args); I don't think I can use shellescape for unless I'm missing something.


<!-- Sent from Reviewable.io -->
